### PR TITLE
Document how to document

### LINKED ISSUE #16477: A document for how to write documents
We should have a document telling people how to write documents. 

For example:
- Use auto-generated munge TOC
- Use auto-synced munge example
- How to write links (use relative links when possible, style as `[Description](link)`)
- etc

cc @bgrant0607

### PR DESCRIPTION
**Pull request by janetkuo**
Fixes #16477

cc @kubernetes/goog-ux @eparis @mikedanese

### CODE REVIEW

File: docs/devel/how-to-doc.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+At the beginning of each document, you should insert the following mungers, which warn the users when they're visiting the HEAD.
+
+```
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
```
Comment (eparis) [inline docs/devel/how-to-doc.md]: Is this strictly required? I thought the munger would auto-add it if it was missing from any new docs. Same with analytics. You should talk about what this munger does in 1.1 docs and why it matters.

Comment (janetkuo) [inline docs/devel/how-to-doc.md]: Oh, yes, they're auto-added if missing. I'll move both to an "optional" or "auto-added mungers" section.

Comment (janetkuo) [inline docs/devel/how-to-doc.md]: Updated. PTAL.

**Comment by janetkuo** [comment]
Just found PR #11568 absolutizes links from /docs to anywhere but examples/ or docs/. I'll add this rule in this PR too. cc @bgrant0607 @davidopp

File: docs/devel/how-to-doc.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    ports:
+    - containerPort: 80
+```
+
+[Download example](../user-guide/pod.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE ../user-guide/pod.yaml -->
+
+## Adding Links
+
+Use inline link instead of url at all times. When you add internal links from `docs/` to `docs/` or `examples/`, use relative links; otherwise, use `http://releases.k8s.io/HEAD/<path/to/link>`. For example, use:
```
Comment (janetkuo) [inline docs/devel/how-to-doc.md]: Added this based on #11568. cc @davidopp @bgrant0607

File: docs/devel/how-to-doc.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+## What Are Mungers?
+
+Mungers are like gofmt for md docs which we use to format documents. To use it, simply place
+
+```
+<!-- BEGIN MUNGE: xxxx -->
+<!-- END MUNGE: xxxx -->
+```
+
+in your md files. Note that xxxx is the placeholder for a specific munger. Appropriate content will be generated and inserted between two brackets after you run `hack/update-generated-docs.sh`. See [munger document](../../cmd/mungedocs/) for more details.
```
Comment (bgrant0607) [inline docs/devel/how-to-doc.md]: This link violates the rule about absolute links to non-doc directories. :-)

File: docs/devel/how-to-doc.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    - [Generate Analytics](#generate-analytics)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## What Are Mungers?
+
+Mungers are like gofmt for md docs which we use to format documents. To use it, simply place
+
+```
+<!-- BEGIN MUNGE: xxxx -->
```
Comment (bgrant0607) [inline docs/devel/how-to-doc.md]: One only needs to know this when writing a new munger. I'd put this at the end of the doc. Most people need to know:

1. That we munge the docs and how to do it (update-generated-docs.sh)
2. That we verify the docs have been munged
3. How to write links
4. How to get a TOC
5. How to include an example
6. Other misc. doc conventions, such as when to use triple vs. single backticks, console vs. shell quote format, whether to use one `#` for the doc title or equal signs, etc.

Comment (janetkuo) [inline docs/devel/how-to-doc.md]: Fixed

**Comment by janetkuo** [comment]
PTAL

File: docs/devel/how-to-doc.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+Wrap a span of code with single backticks (`` ` ``). To format multiple lines of code as its own code block, use triple backticks (```` ``` ````).
+
+### Syntax Highlighting
+
+Adding syntax highlighting to code blocks improves readability. To do so, in your fenced block, add an optional language identifier. Some useful identifier includes `yaml`, `console` (for console output), and `sh` (for shell quote format). Note that in a console output, put `$ ` at the beginning of each command and put nothing at the beginning of the output. Here's an example of console code block:
+
+```
+```console
+
```
Comment (eparis) [inline docs/devel/how-to-doc.md]: useless whitespace.

**Comment by eparis** [comment]
(If you choose to update the whitespace, feel free to re-add the LGTM)

**Pull request merged**